packages/build/src/plugins_core/deploy_config/index.ts

```diff
@@ -0,0 +1,87 @@
+import { promises as fs } from 'fs'
+import { resolve } from 'path'
+
+import merge from 'deepmerge'
+
+import type { NetlifyConfig } from '../../index.js'
+import { CoreStep, CoreStepFunction } from '../types.js'
+
+import { filterConfig } from './util.js'
+
+const ALLOWED_PROPERTIES = [
+  ['build', 'edge_functions'],
+  ['build', 'functions'],
+  ['build', 'publish'],
+  ['functions', '*'],
+  ['functions', '*', '*'],
+  ['headers'],
+  ['images', 'remote_images'],
+  ['redirects'],
+]
+
+const OVERRIDE_PROPERTIES = new Set(['headers!', 'redirects!'])
+
+const coreStep: CoreStepFunction = async function ({
+  buildDir,
+  netlifyConfig,
+  systemLog = () => {
+    // no-op
+  },
+}) {
+  const configPath = resolve(buildDir, '.netlify', 'deploy', 'v1', 'config.json')
```

Comment:
NIT: easier to search and find for paths if we preserve paths

```suggestion
  const configPath = resolve(buildDir, '.netlify/deploy/v1/config.json')
```

Reply:
Done.